### DISCUSSION TITLE
WiX installer create custom PuTTY profile which includes default settings

### DISCUSSION THREAD

**Discussion by benomalley25** [wixtoolset/issues#6921] [Questions]
I have an app with a wix built installer that, during install, creates several different putty session profiles by adding registry values, similar to below:
`<RegistryKey Root="HKCU"
           Key="Software\\SimonTatham\\PuTTY\\Sessions\\\<custom-profile\>"
    Action="create"\>
      \<RegistryValue Type="string" Name="Colour0" Value="210,210,210"/\>
  \</RegistryKey\>`

This is working well, but I would like to make it so that if there are any existing PuTTY default settings these are copied into these new profiles on install. I found this tutorial for copying registry values in WiX (https://subscription.packtpub.com/book/application-development/9781849513722/10/ch10lvl1sec63/copying-registry-values) but this is limited to individual values, and I'd like to copy over EVERY value from default settings (that I'm not explicitly overwriting).

I tried doing a registrysearch targetting the directory:
`<Property Id="DEFAULT_PUTTY_CONFIG">
          <RegistrySearch Id="MyRegistrySearch"
                          Root="HKCU"
                          Key="Software\SimonTatham\PuTTY\Sessions"
                          Name="Default%20Settings"
                          Type="directory" />
        </Property>`

and then inserting it with:
`<RegistryValue Id="CopiedDefaultValues"
                                    Name="CopiedDefaultValues"
                                    Value="[DEFAULT_PUTTY_CONFIG]"
                                    Action="append"
                                    Type="directory" />`

but unfortunately this hasn't worked, it's just dropping a DWORD value named "CopiedDefaults" into my new profile with a value of 1.

Any help would be appreciated.

**Comment by robmen** [comment]
Enumerating an unknown set of registry keys requires a Custom Action. You can write the code to read the keys and add temporary rows to the MSI database to let the Windows Installer do the heavy lifting of install, rollback, etc.
